Add a script for running salt

mattmb opened the pull request:
This adds a salt_utils module which can run salt on the master, gather and parse the results to decide if state execution succeeded. Otherwise running a salt state always produces a 0 error code.

I've deliberately not included salt as a requirement seeing as this will only run on the remote machines where salt will already be installed.

Commit:
Add a script for running salt

This adds a salt_utils module which can run salt on the master,
gather and parse the results to decide if state execution succeeded.
Otherwise running a salt state always produces a 0 error code.

I've modified the bootstrap script to clone this repo so that the
new module is available to be called from fabric during deploy.

kyriakosoikonomakos commented on bootstrap_cfn/salt_utils.py:
```diff
@@ -0,0 +1,49 @@
+    result = r.cmd('jobs.lookup_jid', [jid])
+    if result:
+        return result
+    return False
```
This could be writte as "return result if result else False" and get rid lines 25, 26

stuartornum replied:
it's easier to read as it is

kyriakosoikonomakos replied:
just saying :)

kyriakosoikonomakos commented on bootstrap_cfn/salt_utils.py:
```diff
@@ -0,0 +1,49 @@
+
+def check_state_result(result):
+    results = []
+    for minon, ret in result.items():
```
typo, you probably meant minion

mattmb replied:
I did and I'm not using it anyway so result.values() would be better

kyriakosoikonomakos replied:
Yes.

kyriakosoikonomakos commented:
I don't see where is this called ?

mattmb commented:
@kyriakosoikonomakos see https://github.com/ministryofjustice/template-deploy/issues/2

Commit:
Command line opts for salt_utils

Add command line options for salt utils. This makes running the
salt utils remotely more straightforward.
